### ISSUE TITLE
Regression with cwd keymap 

### ISSUE THREAD

**Issue by ranebrown** [nvim-telescope/telescope-file-browser.nvim#137]
### Description

I'm seeing a regression from #134 / 442f8f1b4faa9307dca59ef48a3a42d9ec75fa9b which breaks search/filtering.

### Neovim version

```markdown
NVIM v0.8.0-dev+9e4286294
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

macOS 12.3.1

### Steps to reproduce

1. Set keymap
  ```lua
  vim.keymap.set("n", "-", function()
      telescope.extensions.file_browser.file_browser({ cwd = vim.fn.expand("%:p:h") })
  end)
  ```
2. Open a file within nested directory structure. Press `-` to open file browser. Press `-` again to jump up one level.
3. Attempt to type a folder or file name and matches will be removed / not found.

### Expected behavior

Filtering/searching works correctly.

### Actual behavior

After jumping up an additional level filtering no longer works.

### Minimal config

```Lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvim/site]]
local package_root = '/tmp/nvim/site/pack'
local install_path = package_root .. '/packer/start/packer.nvim'
local function load_plugins()
  require('packer').startup {
    {
      'wbthomason/packer.nvim',
      {
        'nvim-telescope/telescope.nvim',
        requires = {
          'nvim-lua/plenary.nvim',
          'nvim-telescope/telescope-file-browser.nvim',
        },
      },
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. '/plugin/packer_compiled.lua',
      display = { non_interactive = true },
    },
  }
end
_G.load_config = function()
  require('telescope').setup()
  require('telescope').load_extension('file_browser')
  -- ADD INIT.LUA SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing Telescope and dependencies.")
  vim.fn.system { 'git', 'clone', '--depth=1', 'https://github.com/wbthomason/packer.nvim', install_path }
end
load_plugins()
require('packer').sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua load_config()]]
```

**Comment by fdschmidt93** [comment]
Thanks for reporting, your repro steps are not fully complete (`-` is not mapped by default to `fb_actions.goto_parent_dir`), but I figured out what you meant and what went wrong :sweat_smile: 

This should now be fixed with #138 Please let me know in case you have more problems.

**Comment by ranebrown** [comment]
Thanks! Glad you figured it out without the additional setup, sorry about that.
